lte/gateway/c/session_manager/LocalEnforcer.cpp

```diff
@@ -2004,6 +2004,32 @@ void LocalEnforcer::remove_rule_due_to_bearer_creation_failure(
       static_rule_to_remove, dynamic_rule_to_remove, RequestOriginType::GX);
 }
 
+std::unique_ptr<Timezone> LocalEnforcer::compute_access_timezone() {
+  if (!SEND_ACCESS_TIMEZONE) {
+    MLOG(MWARNING) << "send_access_timezone field is not set, not computing "
+                      "timezone information";
+    return nullptr;
+  }
+  Timezone timezone;
+  time_t now           = std::time(NULL);
+  auto const localtime = *std::localtime(&now);
+  std::ostringstream os;
+  os << std::put_time(&localtime, "%z");
+  std::string s = os.str();
+  // s is in ISO 8601 format: "±HHMM"
+  if (s.size() < 5) {
+    MLOG(MERROR) << "Failed to set access timezone";
+    return nullptr;
+  }
+  int hours            = std::stoi(s.substr(0, 3), nullptr, 10);
+  int minutes          = std::stoi(s[0] + s.substr(3), nullptr, 10);
+  int total_offset_min = hours * 60 + minutes;
+  MLOG(MINFO) << "Access timezone is UTC " << (total_offset_min >= 0 ? "+" : "")
+              << total_offset_min << " minutes";
+  timezone.set_offset_minutes(total_offset_min);
+  return std::make_unique<Timezone>(timezone);
+}
+
```

Comment:
just curious, but why don't you just send the UTC/localtime time to FEG and leave FEG compute the timezone? This way the translation is all in one place and we may get to use that time for something else (you don't need to change that btw 😅)

Reply:
Oh yes, I suppose I could just do that too :D